### ISSUE TITLE
Column "INFORMATION_SCHEMA.COLUMNS.DOMAIN_SCHEMA" and "INFORMATION_SCHEMA.COLUMNS.DOMAIN_NAME" not found in H2 1.4.197

### ISSUE THREAD

**Issue by Skym0sh0** [jOOQ/jOOQ#12007]
As in issue https://github.com/jOOQ/jOOQ/issues/11733 in H2 1.4.197 there are more columns missing in the INFORMATION_SCHEMA during jOOQ Code Generation:

```
Error while initialising type
org.jooq.exception.DataAccessException: SQL [select INFORMATION_SCHEMA.COLUMNS.COLUMN_NAME, INFORMATION_SCHEMA.COLUMNS.ORDINAL_POSITION, INFORMATION_SCHEMA.COLUMNS.TYPE_NAME TYPE_NAME, INFORMATION_SCHEMA.COLUMNS.COLUMN_TYPE COLUMN_TYPE, case when (INFORMATION_SCHEMA.COLUMNS.NUMERIC_PRECISION = 65535 and INFORMATION_SCHEMA.COLUMNS.NUMERIC_SCALE = 32767) then 0 else INFORMATION_SCHEMA.COLUMNS.CHARACTER_MAXIMUM_LENGTH end CHARACTER_MAXIMUM_LENGTH, decode(INFORMATION_SCHEMA.COLUMNS.NUMERIC_PRECISION, 65535, 0, INFORMATION_SCHEMA.COLUMNS.NUMERIC_PRECISION) NUMERIC_PRECISION, decode(INFORMATION_SCHEMA.COLUMNS.NUMERIC_SCALE, 32767, 0, INFORMATION_SCHEMA.COLUMNS.NUMERIC_SCALE) NUMERIC_SCALE, INFORMATION_SCHEMA.COLUMNS.IS_NULLABLE, INFORMATION_SCHEMA.COLUMNS.COLUMN_DEFAULT, INFORMATION_SCHEMA.COLUMNS.REMARKS, INFORMATION_SCHEMA.COLUMNS.SEQUENCE_NAME, INFORMATION_SCHEMA.COLUMNS.DOMAIN_SCHEMA, INFORMATION_SCHEMA.COLUMNS.DOMAIN_NAME from INFORMATION_SCHEMA.COLUMNS where (INFORMATION_SCHEMA.COLUMNS.TABLE_SCHEMA = cast(? as varchar(2147483647)) and INFORMATION_SCHEMA.COLUMNS.TABLE_NAME = cast(? as varchar(2147483647))) order by INFORMATION_SCHEMA.COLUMNS.ORDINAL_POSITION]; Feld "INFORMATION_SCHEMA.COLUMNS.DOMAIN_SCHEMA" nicht gefunden
Column "INFORMATION_SCHEMA.COLUMNS.DOMAIN_SCHEMA" not found; SQL statement:
select INFORMATION_SCHEMA.COLUMNS.COLUMN_NAME, INFORMATION_SCHEMA.COLUMNS.ORDINAL_POSITION, INFORMATION_SCHEMA.COLUMNS.TYPE_NAME TYPE_NAME, INFORMATION_SCHEMA.COLUMNS.COLUMN_TYPE COLUMN_TYPE, case when (INFORMATION_SCHEMA.COLUMNS.NUMERIC_PRECISION = 65535 and INFORMATION_SCHEMA.COLUMNS.NUMERIC_SCALE = 32767) then 0 else INFORMATION_SCHEMA.COLUMNS.CHARACTER_MAXIMUM_LENGTH end CHARACTER_MAXIMUM_LENGTH, decode(INFORMATION_SCHEMA.COLUMNS.NUMERIC_PRECISION, 65535, 0, INFORMATION_SCHEMA.COLUMNS.NUMERIC_PRECISION) NUMERIC_PRECISION, decode(INFORMATION_SCHEMA.COLUMNS.NUMERIC_SCALE, 32767, 0, INFORMATION_SCHEMA.COLUMNS.NUMERIC_SCALE) NUMERIC_SCALE, INFORMATION_SCHEMA.COLUMNS.IS_NULLABLE, INFORMATION_SCHEMA.COLUMNS.COLUMN_DEFAULT, INFORMATION_SCHEMA.COLUMNS.REMARKS, INFORMATION_SCHEMA.COLUMNS.SEQUENCE_NAME, INFORMATION_SCHEMA.COLUMNS.DOMAIN_SCHEMA, INFORMATION_SCHEMA.COLUMNS.DOMAIN_NAME from INFORMATION_SCHEMA.COLUMNS where (INFORMATION_SCHEMA.COLUMNS.TABLE_SCHEMA = cast(? as varchar(2147483647)) and INFORMATION_SCHEMA.COLUMNS.TABLE_NAME = cast(? as varchar(2147483647))) order by INFORMATION_SCHEMA.COLUMNS.ORDINAL_POSITION [42122-197]
	at org.jooq_3.14.11.H2.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2903)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:757)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:389)
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:337)
	at org.jooq.impl.AbstractResultQuery.iterator(AbstractResultQuery.java:348)
	at org.jooq.impl.SelectImpl.iterator(SelectImpl.java:2890)
	at org.jooq.meta.h2.H2TableDefinition.getElements0(H2TableDefinition.java:92)
	at org.jooq.meta.AbstractElementContainerDefinition.getElements(AbstractElementContainerDefinition.java:88)
	at org.jooq.meta.AbstractTableDefinition.getColumns(AbstractTableDefinition.java:172)
	at org.jooq.codegen.JavaGenerator.generateTable(JavaGenerator.java:5025)
	at org.jooq.codegen.JavaGenerator.generateTables(JavaGenerator.java:5012)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:582)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:537)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:436)
	at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:879)
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:233)
```

As I researched further, this holds also for column `DOMAIN_NAME`.

**Comment by lukaseder** [comment]
Thanks a lot for your report. We currentl ydon't have versioned dialect support for H2: https://github.com/jOOQ/jOOQ/issues/11737

H2 has been moving fast and breaking a few things in the recent releases. The improvements were *very* significant, which definitely justifies breaking compatibility here and there. Given how H2 is mostly being used (e.g. as a test database), I didn't think there would be a lot of value to supporting older versions of H2, but we'll definitely add versions to support both 1.4.200 and 2.0 (once it's out).

What would be a use-case for still using 1.4.197?

**Comment by Skym0sh0** [comment]
As in #11733 we are stuck due to our schema evolution which prevents us from updating to H2 1.4.198 or higher. We plan to do some kind of schema squashing to get rid of the problems we are suffering with H2.

Would it be possible to do some workaround like https://github.com/jOOQ/jOOQ/issues/11733#issuecomment-810285837 ?

**Comment by lukaseder** [comment]
Thanks for the pointer. I already forgot about that other issue :)

> Would it be possible to do some workaround like [#11733 (comment)](https://github.com/jOOQ/jOOQ/issues/11733#issuecomment-810285837) ?

Perhaps, but I guess these kinds of issues will appear all the time, if we don't set up any infrastructure for testing jOOQ with H2 1.4.197 for you, and I'm not convinced this should be a priority for the jOOQ Open Source Edition.

Are you licensed? If not, would you be interested in getting licensed if I prioritised https://github.com/jOOQ/jOOQ/issues/11737 ? There would be an H2 1.4.197 and H2 1.4.200 dialect (for now), and we'd guarantee support for these two versions in the jOOQ Professional Edition...

**Comment by Skym0sh0** [comment]
Yes, we are licensed as our production database is Oracle.

**Comment by lukaseder** [comment]
Perfect, I'll see what can be done tomorrow. With tests in place, we can make sure this won't break again.

Can you just send a quick mail to contact@datageekery.com with your license key, for verification?

**Comment by lukaseder** [comment]
Thanks a lot for the verification. I'll expedite support for versioned H2 dialects under the jOOQ Professional Edition via https://github.com/jOOQ/jOOQ/issues/11737.

For now, we'll support 1.4.197 and 1.4.200. We'll run integration tests for these code generation queries and for the runtime library. I'll document any possible additional issues that might be found here or in #11737.

The change will be included in jOOQ 3.15, which is due this month if there are no major delays. While jOOQ 3.14 won't include the new versioned dialect, some fixes can still be backported, just like #11733.

Please let me know if you need anything else at this point.

**Comment by lukaseder** [comment]
Looking into this now. It seems our integration tests are running a query that corrupts the H2 database after which all tests fail with a NPE... A lot seems to have been improved since 1.4.197...

Anyway, we'll skip the failing tests and make sure the `jOOQ-meta` queries work well, which is the main goal for now.

**Comment by lukaseder** [comment]
Integration tests show that the query in question works in version 1.4.198. Will avoid accessing the `DOMAIN_NAME` and `DOMAIN_SCHEMA` columns for 1.4.197 and less.

**Comment by lukaseder** [comment]
Alright. Integration tests are now running to (hopefully) prevent further regressions in `jOOQ-meta` for H2 1.4.197.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.12 (https://github.com/jOOQ/jOOQ/issues/12039)

**Comment by lukaseder** [comment]
3.14.12 has been released including this fix
